guard/src/commands/parse_tree.rs

```diff
@@ -1,65 +1,33 @@
-use crate::command::Command;
-use crate::commands::{OUTPUT, PARSE_TREE, PRINT_JSON, PRINT_YAML, RULES};
+use crate::commands::{Executable, PRINT_JSON, PRINT_YAML, SUCCESS_STATUS_CODE};
 use crate::rules::Result;
 use crate::utils::reader::Reader;
 use crate::utils::writer::Writer;
-use clap::{Arg, ArgAction, ArgMatches, ValueHint};
+use clap::Args;
 use std::fs::File;
 
-#[derive(Clone, Copy, Eq, PartialEq)]
-pub struct ParseTree {}
-
-#[allow(clippy::new_without_default)]
-impl ParseTree {
-    pub fn new() -> Self {
-        ParseTree {}
-    }
+const ABOUT: &str = "Prints out the parse tree for the rules defined in the file.";
+const OUTPUT_HELP: &str = "Write to output file";
+const PRINT_JSON_HELP: &str = "Print output in JSON format. Use -p as the short flag";
+const PRINT_YAML_HELP: &str = "Print output in YAML format";
+const RULES_HELP: &str = "Provide a rules file";
+
+#[derive(Debug, Clone, Eq, PartialEq, Args)]
+#[clap(about=ABOUT)]
+#[clap(arg_required_else_help = true)]
+pub(crate) struct ParseTree {
+    #[arg(short, long, help = RULES_HELP)]
+    pub(crate) rules: Option<String>,
+    #[arg(short, long, help = OUTPUT_HELP)]
+    pub(crate) output: Option<String>,
+    #[arg(short=PRINT_JSON.1, long=PRINT_JSON.0, help=PRINT_JSON_HELP)]
+    pub(crate) print_json: bool,
+    #[arg(short=PRINT_YAML.1, long=PRINT_YAML.0, help=PRINT_YAML_HELP)]
+    pub(crate) print_yaml: bool,
 }
 
-impl Command for ParseTree {
-    fn name(&self) -> &'static str {
-        PARSE_TREE
-    }
-
-    fn command(&self) -> clap::Command {
-        clap::Command::new(PARSE_TREE)
-            .about("Prints out the parse tree for the rules defined in the file.")
-            .arg(
-                Arg::new(RULES.0)
-                    .long(RULES.0)
-                    .short(RULES.1)
-                    .help("Provide a rules file")
-                    .value_hint(ValueHint::FilePath)
```

Comment:
Are we losing these intentionally removing these hints / does it matter they're removed here?

Reply:
from manual testing they didnt seem to provide any value with the completions script